library/common.h

```diff
@@ -318,6 +318,21 @@ static inline void mbedtls_xor_no_simd(unsigned char *r,
 #define MBEDTLS_UNLIKELY(x)     x
 #endif
 
+#if defined(__has_builtin)
+#if __has_builtin(__builtin_assume)
+/* MBEDTLS_ASSUME may be used to provide additional information to the compiler
+ * which can result in smaller code-size. */
+#define MBEDTLS_ASSUME(x)       __builtin_assume(x)
+// clang provides __builtin_assume
```

Comment:
Are we allowed to use C++-style `//` comments?

Reply:
Yes, but tidied up for consistency.

Reply:
They are part of the C99 subset that we use, and were even part of the C90 dialect that we allowed back before we switched to C99.

Reply:
It's weird - we use them liberally for commenting-out config options, but very rarely in the code itself